### DISCUSSION TITLE
Are there instructions for running the unit tests to contribute to the xunit.runner.visualstudio project?

### DISCUSSION THREAD

**Discussion by jtreher** [xunit/xunit#2658] [Question] [answered]
Hello,

I wanted to update the packages to address dependabot PRs as well as update to .net 6.0 since netcore3.1 is EOL (my commit: https://github.com/xunit/visualstudio.xunit/compare/main...jtreher:visualstudio.xunit:main). I looked at the readme and contributor guide, but I don't quite understand what steps need to be taken before I can run the tests. I see there is some documentation on debugging it, but I'm only trying to get the test to run at this point.

Thanks,

Justin

**Comment by bradwilson** [comment, marked as answer]
Thanks, you're right that the README does not contain instructions on how to build the project or run the tests. I've added an issue to get this documented: https://github.com/xunit/visualstudio.xunit/issues/361

The test project (`test.xunit.runner.visualstudio`) should be runnable from within Test Explorer. I just pushed up a few changes for dependencies and a broken test, and verified that they work for me with VS 2022 version 17.4.4:

![image](https://user-images.githubusercontent.com/16944/216272423-183b6145-10a0-4169-a260-9eed0a009748.png)

**Comment by jtreher** [comment]
Excellent! So, the only test project that's supposed to run is the text.xunit.runner.visualstudio and the others are not actually meant be run directly, but come along as part of running the main test? I was trying to run them all.

Also, would it be possible to upgrade the framework to target `net6.0` instead of `netcoreapp3.1` or did you leave that out intentionally?

Yep, all green.
![image](https://user-images.githubusercontent.com/1258179/216318674-75c25c1f-d3c9-46ee-bf6c-925e4cfd421c.png)

**Reply by bradwilson** [reply]
The project is owned by @clairernovotny so she will need to be the one to answer on what should be runnable or not, and when/whether the upgrade from netcoreapp3.1 to net6.0 will take place.
